service/history/historyReplicator.go

```diff
@@ -111,11 +114,17 @@ func (r *historyReplicator) ApplyEvents(ctx context.Context, request *h.Replicat
 		logging.TagWorkflowExecutionID: request.WorkflowExecution.GetWorkflowId(),
 		logging.TagWorkflowRunID:       request.WorkflowExecution.GetRunId(),
 		logging.TagSourceCluster:       request.GetSourceCluster(),
-		logging.TagVersion:             request.GetVersion(),
+		logging.TagIncomingVersion:     request.GetVersion(),
 		logging.TagFirstEventID:        request.GetFirstEventId(),
 		logging.TagNextEventID:         request.GetNextEventId(),
 	})
 
+	r.metricsClient.RecordTimer(
```

Comment:
It seems strange to report replication events size as a timer which is generally used for computing latencies.  Is the intention here to compute various percentiles for this counter?

Reply:
yes, i would like to get the distribution of the number of events per replication task, this can be useful for debugging or explaining the potential high latency issue